### PR TITLE
Fixes related to tests and upgrade go version to 1.17

### PR DESCRIPTION
**Pull request by revit13**
This PR contains the following changes:

- upgrade go version to 1.17
- fix test target: as the test requires k8s client to read the secrets, it creates a cluster before running the tests
- adds docker related targets to the makefile (although goreleaser workflow actually pushes the docker image upon release)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM alpine:latest
+
+WORKDIR /
+COPY vault/plugins/vault-plugin-secrets-kubernetes-reader .
+
```
Comment (roytman) [inline Dockerfile:5]: add `USER 10001`

File: Makefile [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
 .DEFAULT_GOAL := all
 
+DOCKER_HOSTNAME ?= ghcr.io
+DOCKER_NAMESPACE ?= fybrik
+DOCKER_TAG ?= 0.0.0
+DOCKER_NAME ?= vault-plugin-secrets-kubernetes-reader
+
+IMG := ${DOCKER_HOSTNAME}/${DOCKER_NAMESPACE}/${DOCKER_NAME}:${DOCKER_TAG}
+
 all: build
```
Comment (roytman) [inline Makefile]: in fybrik we call it `source-build`

File: go.mod [hunk 3]
```diff
@@ -1,13 +1,76 @@
...
 
-go 1.13
+go 1.17
 
 require (
 	github.com/hashicorp/go-hclog v0.15.0
 	github.com/hashicorp/vault/api v1.0.4
 	github.com/hashicorp/vault/sdk v0.1.13
+	github.com/onsi/gomega v1.20.1
 	k8s.io/api v0.20.4
```
Comment (roytman) [inline go.mod]: should it be v0.23.5 ?

File: go.mod [hunk 4]
```diff
@@ -1,13 +1,76 @@
...
 	k8s.io/apimachinery v0.20.4
```
Comment (roytman) [inline go.mod]: k8s.io/apimachinery v0.23.5

File: go.mod [hunk 5]
```diff
@@ -1,13 +1,76 @@
...
 	k8s.io/client-go v0.20.2
```
Comment (roytman) [inline go.mod]: k8s.io/client-go v0.23.5

File: go.mod [hunk 6]
```diff
@@ -1,13 +1,76 @@
...
 	sigs.k8s.io/controller-runtime v0.8.2
```
Comment (roytman) [inline go.mod]: sigs.k8s.io/controller-runtime v0.11.2

File: hack/make-rules/tools.mk [hunk 7]
```diff
@@ -0,0 +1,23 @@
+INSTALL_TOOLS += $(TOOLBIN)/kind
+$(TOOLBIN)/kind:
+	GOBIN=$(ABSTOOLBIN) go install sigs.k8s.io/kind@v0.11.1
+
+INSTALL_TOOLS += $(TOOLBIN)/yq
+.PHONY: $(TOOLBIN)/yq
+$(TOOLBIN)/yq:
+	cd $(TOOLS_DIR); ./install_yq.sh
+
+INSTALL_TOOLS += $(TOOLBIN)/kubebuilder
```
Comment (roytman) [inline hack/make-rules/tools.mk]: why do you need kubebuilder here?

File: hack/make-rules/tools.mk [hunk 8]
```diff
@@ -0,0 +1,23 @@
+INSTALL_TOOLS += $(TOOLBIN)/kind
+$(TOOLBIN)/kind:
+	GOBIN=$(ABSTOOLBIN) go install sigs.k8s.io/kind@v0.11.1
+
+INSTALL_TOOLS += $(TOOLBIN)/yq
```
Comment (roytman) [inline hack/make-rules/tools.mk:5]: where do you use `yq` ?

File: hack/tools/install_kubebuilder.sh [hunk 9]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (roytman) [inline hack/tools/install_kubebuilder.sh]: we don't need it

File: hack/tools/install_yq.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (roytman) [inline hack/tools/install_yq.sh:1]: do we need it ?

File: hack/tools/requirements.sh [hunk 11]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# Copyright 2020 The Kubernetes Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+set -o nounset
+
+HELM_VERSION=v3.8.2
+ISTIO_VERSION=1.8.0
```
Comment (roytman) [inline hack/tools/requirements.sh]: we don't need istio

File: hack/tools/requirements.sh [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+JSON_SCHEMA_GENERATOR_VERSION=0.2.1
```
Comment (roytman) [inline hack/tools/requirements.sh]: we don't need it

File: hack/tools/requirements.sh [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+KUBE_VERSION=1.22.0
+KUSTOMIZE_VERSION=3.2.0
```
Comment (roytman) [inline hack/tools/requirements.sh]: do we use kustomize ?
and others ?

File: hack/make-rules/tools.mk (hunk 7 above)
Comment (revit13) [reply hack/make-rules/tools.mk]: for kubectl, it is needed for kind target

File: hack/make-rules/tools.mk (hunk 8 above)
Comment (revit13) [reply hack/make-rules/tools.mk:5]: it is needed for kubectl installation. it should be added a dependency to it. please look at install_kubebuilder.sh

File: hack/tools/requirements.sh (hunk 11 above)
Comment (revit13) [reply hack/tools/requirements.sh]: I copied the file as is from fybrik. I'll remove reference fot unneeded tools.

File: Dockerfile (hunk 1 above)
Comment (revit13) [reply Dockerfile:5]: Done, thanks

File: Makefile (hunk 2 above)
Comment (revit13) [reply Makefile]: Changed, thanks

File: go.mod (hunk 3 above)
Comment (revit13) [reply go.mod]: Done, thanks

File: go.mod (hunk 4 above)
Comment (revit13) [reply go.mod]: Done, thanks

File: go.mod (hunk 5 above)
Comment (revit13) [reply go.mod]: Done, thanks

File: go.mod (hunk 6 above)
Comment (revit13) [reply go.mod]: Done, thanks

File: hack/tools/requirements.sh (hunk 12 above)
Comment (revit13) [reply hack/tools/requirements.sh]: Fixed. Thanks

File: hack/tools/requirements.sh (hunk 13 above)
Comment (revit13) [reply hack/tools/requirements.sh]: I copied the file as is from fybrik. I'll remove references for unneeded tools.

File: hack/tools/install_yq.sh (hunk 10 above)
Comment (revit13) [reply hack/tools/install_yq.sh:1]: it is needed for kubectl installation. it should be added a dependency to it. please look at install_kubebuilder.sh

**Comment by revit13** [comment]
@roytman I changed the tools to include only kubectl and yq which is needed for kubectl script, as you suggested. 
I also made changes based on @cdoron comments.
Thanks

**Review by roytman — APPROVED** [review]

**Pull request merged**